docs/assets/img/docs/managing-pull-requests.drawio.svg

```diff
@@ -0,0 +1,498 @@
+        <path d="M 119 411 L 74.5 411 Q 69 411 69 405.5 L 69 382.5 Q 69 377 74.32 375.62 L 163.68 352.38 Q 169 351 169 356.5 L 169 405.5 Q 169 411 163.5 411 Z" fill="#ffffff" stroke="#000000" stroke-width="2" stroke-miterlimit="10" pointer-events="all"/>
+        <g transform="translate(-0.5 -0.5)">
+            <switch>
+                <foreignObject style="overflow: visible; text-align: left;" pointer-events="none" width="100%" height="100%" requiredFeatures="http://www.w3.org/TR/SVG11/feature#Extensibility">
+                    <div xmlns="http://www.w3.org/1999/xhtml" style="display: flex; align-items: unsafe center; justify-content: unsafe center; width: 98px; height: 1px; padding-top: 387px; margin-left: 70px;">
+                        <div style="box-sizing: border-box; font-size: 0; text-align: center; ">
+                            <div style="display: inline-block; font-size: 12px; font-family: system-ui-monospace,ui-monospace,Lucida Console,Menlo,monospace; color: #000000; line-height: 1.2; pointer-events: all; white-space: normal; word-wrap: normal; ">
+                                <p style="line-height: 120%">
+                                    brew pr-pull
+                                </p>
+                            </div>
+                        </div>
+                    </div>
+                </foreignObject>
+                <text x="119" y="391" fill="#000000" font-family="system-ui-monospace,ui-monospace,Lucida Console,Menlo,monospace" font-size="12px" text-anchor="middle">
+                    brew pr-pull
+                </text>
+            </switch>
+        </g>
+        <path d="M 69 701 L 92 641 L 146 641 L 169 701 Z" fill="#ffffff" stroke="#000000" stroke-width="2" stroke-miterlimit="10" transform="translate(0,671)scale(1,-1)translate(0,-671)" pointer-events="all"/>
+        <g transform="translate(-0.5 -0.5)">
+            <switch>
+                <foreignObject style="overflow: visible; text-align: left;" pointer-events="none" width="100%" height="100%" requiredFeatures="http://www.w3.org/TR/SVG11/feature#Extensibility">
+                    <div xmlns="http://www.w3.org/1999/xhtml" style="display: flex; align-items: unsafe flex-start; justify-content: unsafe center; width: 98px; height: 1px; padding-top: 708px; margin-left: 70px;">
+                        <div style="box-sizing: border-box; font-size: 0; text-align: center; ">
+                            <div style="display: inline-block; font-size: 12px; font-family: Helvetica; color: #000000; line-height: 1.2; pointer-events: all; white-space: normal; word-wrap: normal; ">
+                                Modify and push
+                            </div>
+                        </div>
+                    </div>
+                </foreignObject>
+                <text x="119" y="720" fill="#000000" font-family="Helvetica" font-size="12px" text-anchor="middle">
+                    Modify and push
```

Comment:
This doesn't seem quite right here; `brew pr-pull` will publish all bottles by default so shouldn't be what you push and repeat. Should this be `hub checkout` or similar?

Reply:
No idea, unfortunately.
@SMillerDev I took the `brew pr-pull` verbatim from that chart you’ve made for me. Can you clarify?

Reply:
I think these are two different use cases.

One is that there is a minor change to make to the PR before completing it that won't require running CI or rebuilding bottles (e.g. rewording the commit or adding a comment). In this case, it's appropriate to use `brew pr-pull` (to publish the bottles and create the bottle commit). Then you make the changes and push to the `master` branch.

The other is if, as a maintainer, you want to push your own changes, not to `master`, but to the PR branch (e.g. the PR author is unresponsive and there's a quick fix that you can make to get a formula working). In this case, you want CI to run again and create new bottles, so you would want to use `hub checkout` or `gh pr checkout` and then push back to the PR branch (not `master`).

We definitely should have the first case (`brew pr-pull`) documented because it's a unique to Homebrew. I think the second case (`hub checkout` or `gh pr checkout`) is less important because it's the same workflow that a maintainer for any GitHub project would follow and probably happens less often. Plus, it doesn't relate directly to this diagram because this would be used for PRs that aren't in a `ready-to-merge` state.

Reply:
My suggestion would be to have this workflow (hopefully this makes sense):

```
Want to modify or rebase?
- Yes: do changes require rerunning CI or rebuilding bottles?
    - Yes: `hub checkout` or `gh pr checkout`, make changes, and push to PR branch. Then go back to `Want to modify or rebase?` step
    - No: `brew pr-pull` (automatically publishes bottles and makes bottle commit), make changes, push to `master` and automatically close the PR. Done!
- No: continue to `Existing formula?` step
```